autoextract/aio/retry.py

```diff
@@ -113,14 +127,28 @@ def __call__(self, retry_state: RetryCallState) -> bool:
             return self.stop_on_network_error(retry_state)
         elif _is_server_error(exc):
             return self.stop_on_server_error(retry_state)
+        elif _is_retriable_query_error(exc):
+            return (
+                self.stop_on_retriable_query_error |
+                stop_after_attempt(exc.max_retries)
+            )(retry_state)
         else:
             raise RuntimeError("Invalid retry state exception: %s" % exc)
 
 
+def _exception_factory(fut):
```

Comment:
Not sure about the `QueryRetryError` and this method. It confused me more than help me when reading the code. Another exception to understand `QueryRetryError` which is different than `QueryError`.

Reply:
This is supposed to differentiate between RetryErrors that happen because of Query-level error versus the ones that happen because of Request-level and Network errors. This was a suggestion from @ejulio to simplify the logic of the request method.

Reply:
Documenting: we've discussed this comment on another channel and you've agreed that this is a minor thing that could be archived for now. 🤝